### PR TITLE
Added method for spack to singularity file

### PR DESCRIPTION
**Pull request by munishchouhan**
This change will convert spack file to singularity file

### CODE REVIEW

File: wave-utils/src/test/groovy/io/seqera/wave/util/DockerHelperTest.groovy [hunk 1]
```diff
@@ -510,4 +510,27 @@ class DockerHelperTest extends Specification {
                     apt-get update -y && apt-get install -y procps
                 '''.stripIndent()
     }
+
+    def 'should create singularity content from spack file'(){
+        given:
+        def SPACK_OPTS = [ commands:['USER hola']]
```
Comment (jordeu) [inline wave-utils/src/test/groovy/io/seqera/wave/util/DockerHelperTest.groovy:516]: Not really related to this PR that everything seems correct, but @pditommaso I've some doubts about this `commands` parameter. 

Where this comes from? Is the user that can add custom commands to the recipe? then, unless we add the prefix `RUN` when building the `Dockerfile`, these commands are not container engine agnostic and they should be different if they're using Singularity or Docker.

**Review by jordeu — APPROVED** [review]
PR seems ok. Only I've some concerns and doubts more generals about the `commands` parameter.

File: wave-utils/src/test/groovy/io/seqera/wave/util/DockerHelperTest.groovy (hunk 1 above)
Comment (jordeu) [reply wave-utils/src/test/groovy/io/seqera/wave/util/DockerHelperTest.groovy:516]: Do we want to make them engine agnostic? or is that ok?

Comment (pditommaso) [reply wave-utils/src/test/groovy/io/seqera/wave/util/DockerHelperTest.groovy:516]: the `commands` is the setting provided by the user via the option `wave.build.spack.command`. It's not expected to be portable across different container engines 

https://www.nextflow.io/docs/edge/wave.html#advanced-settings

**Comment by pditommaso** [comment]
Was the entire loop tested? Likely it's simpler via the Wavelit cli

**Comment by pditommaso** [comment]
Add this project to Wavelit using `includeBuild`, see 

https://github.com/seqeralabs/wavelit/blob/856fc91db237bb2d24f1e34599d95d95b1235c98/settings.gradle#L20

Then submit the request via Wavelit cli to local Wave

**Comment by munishchouhan** [comment]
Test using wavelit:
```
wavelit -s --spack-package cosway --spack-run-command='cowsay Hello World' --wave-endpoint https://09f269899ded.ngrok.app --freeze --build-repo docker.io/hrma017/dev  --tower-token  eyJ0aWQiOiA5fS4xMDNiOTEwMTlkZWYxMzU4MTIzNDQwYjQ3Yjk2MDhiNmE1YzdjMzMw --tower-endpoint http://localhost:8000/api

oras://docker.io/hrma017/dev:224134b04a4cd790d30e70a6f320811e
```

Error in Wave:

```
java.lang.IllegalStateException: Missing Spack cacheMountPath configuration setting
	at io.seqera.wave.configuration.SpackConfig.getCacheMountPath(SpackConfig.groovy:70)
	at io.seqera.wave.service.builder.ContainerBuildServiceImpl.containerFile0(ContainerBuildServiceImpl.groovy:127)
	at io.seqera.wave.service.builder.ContainerBuildServiceImpl.launch(ContainerBuildServiceImpl.groovy:148)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1268)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1035)
	at org.codehaus.groovy.runtime.InvokerHelper.invokePogoMethod(InvokerHelper.java:1036)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethod(InvokerHelper.java:1019)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethodSafe(InvokerHelper.java:97)
	at io.seqera.wave.service.builder.ContainerBuildServiceImpl$_launchAsync_lambda1.doCall(ContainerBuildServiceImpl.groovy:206)
```

**Comment by munishchouhan** [comment]
After setting up the Spack config, Now there is error in building container:
```
15:26:41.405 [wave-builder-1] INFO  i.s.w.s.b.ContainerBuildServiceImpl - == Build completed with status=255; stdout: (see below)
 Unable to find image 'quay.io/singularity/singularity:v3.11.4-slim' locally
 v3.11.4-slim: Pulling from singularity/singularity
 4db1b89c0bd1: Pulling fs layer
 627edb5d4276: Pulling fs layer
 8c1f329c7ae9: Pulling fs layer
 4db1b89c0bd1: Download complete
 4db1b89c0bd1: Pull complete
 8c1f329c7ae9: Verifying Checksum
 8c1f329c7ae9: Download complete
 627edb5d4276: Verifying Checksum
 627edb5d4276: Download complete
 627edb5d4276: Pull complete
 8c1f329c7ae9: Pull complete
 Digest: sha256:6afef0d0dfaa8f0d263120ecbd08ff218d8bdbfe282258e979e0e3db807c042c
 Status: Downloaded newer image for quay.io/singularity/singularity:v3.11.4-slim
 FATAL:   Unable to build from /Users/munish.chouhan/testing_ground/wave/build-workspace/af3d942649b3be91/Containerfile: while parsing definition: /Users/munish.chouhan/testing_ground/wave/build-workspace/af3d942649b3be91/Containerfile: failed to parse deffile header: invalid header keyword found: from spack/ubuntu-jammy
```

Wavelit command:
```
munish.chouhan@192 wavelit %  wavelit -s --spack-package cosway --spack-run-command='cowsay Hello World' --wave-endpoint https://caaa422f6957.ngrok.app --freeze --build-repo docker.io/hrma017/dev  --tower-token  eyJ0aWQiOiA5fS4xMDNiOTEwMTlkZWYxMzU4MTIzNDQwYjQ3Yjk2MDhiNmE1YzdjMzMw --tower-endpoint http://localhost:8000/api
oras://docker.io/hrma017/dev:cosway--af3d942649b3be91
```

**Review by pabloaledo — APPROVED** [review]

**Pull request merged**